app/controllers/users_controller.rb

```diff
@@ -1,11 +1,48 @@
 class UsersController < ApplicationController
   def index
-    @users = User.all # Assuming you have a User model
-    # Additional logic if needed
+    @users = User.all
   end
 
   def show
-    @user_id = params[:id]
-    # Placeholder action for the 'users/:id' URL
+    @user = User.find(params[:id])
+    @posts = @user.three_recent_posts
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to @user
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to @user
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user = User.find(params[:id])
+    @user.destroy
+    redirect_to users_path
+  end
```

Comment:
**_I really appreciate your efforts for defining all actions 👏👏 I noticed that the PR includes a code that is out of the scope of the current project and it is not going to be reviewed 📁Please remember to provide a link to the additional code in the Pull Request created for the next project. For example: you can create a link(s) to specific commit(s) and/or line(s) of code in the Pull Request for current milestone and include that link in the description of Pull Request for your next milestone. Similarly for the `posts_controller`_**